radix-engine/src/blueprints/resource/auth_zone.rs

```diff
@@ -206,26 +151,59 @@ impl AuthZoneBlueprint {
                 TYPE_INFO_FIELD_PARTITION => ModuleInit::TypeInfo(TypeInfoSubstate::Object(ObjectInfo {
                     blueprint: Blueprint::new(&RESOURCE_PACKAGE, NON_FUNGIBLE_PROOF_BLUEPRINT),
                     global: false,
-                    outer_object: Some(input.resource_address.into()),
+                    outer_object: Some(resource_address.into()),
                     instance_schema: None,
                 })).to_substates()
             ),
         )?;
 
-        Ok(IndexedScryptoValue::from_typed(&Proof(Own(node_id))))
+        Ok(Proof(Own(node_id)))
     }
 
-    pub(crate) fn clear<Y>(
-        input: &IndexedScryptoValue,
+    pub(crate) fn create_proof_of_all<Y>(
+        resource_address: ResourceAddress,
         api: &mut Y,
-    ) -> Result<IndexedScryptoValue, RuntimeError>
+    ) -> Result<Proof, RuntimeError>
     where
-        Y: ClientApi<RuntimeError>,
+        Y: KernelNodeApi + KernelSubstateApi<SystemLockData> + ClientApi<RuntimeError>,
     {
-        let _input: AuthZoneClearInput = input.as_typed().map_err(|e| {
-            RuntimeError::SystemUpstreamError(SystemUpstreamError::InputDecodeError(e))
-        })?;
+        let auth_zone_handle = api.actor_lock_field(
+            OBJECT_HANDLE_SELF,
+            AuthZoneField::AuthZone.into(),
+            LockFlags::MUTABLE,
+        )?;
+
+        let auth_zone: AuthZone = api.field_lock_read_typed(auth_zone_handle)?;
+        let proofs: Vec<Proof> = auth_zone.proofs.iter().map(|p| Proof(p.0)).collect();
+        let composed_proof = compose_proof_by_amount(&proofs, resource_address, None, api)?;
+
+        let blueprint_name = match &composed_proof {
+            ComposedProof::Fungible(..) => FUNGIBLE_PROOF_BLUEPRINT,
+            ComposedProof::NonFungible(..) => NON_FUNGIBLE_PROOF_BLUEPRINT,
+        };
+        api.field_lock_write_typed(auth_zone_handle, &auth_zone)?;
+
+        let node_id = api.kernel_allocate_node_id(EntityType::InternalGenericComponent)?;
+        api.kernel_create_node(
```

Comment:
Why can we use `new_simple_object` in `FungibleBucket`'s `create_proof_of_amount`, but not here? Is the former missing an outer_object? Or does `new_simple_object` just call a `new` constructor?

I also wonder if some of this logic could live / be commonized in the proof blueprint to reduce the risk of it diverging? :)

Reply:
(appreciate this was just moved in this PR, so happy if we come back to this later)

Reply:
Let's revisit this as part of the "removing dependency on kernel api from native blueprints".